File: android/app/src/main/java/com/neostudios/zest/domain/model/Cookbook.kt
package com.neostudios.zest.domain.model

data class Cookbook(
    val id: String? = null,
    val name: String,
    val description: String? = null,
    val recipeIds: List<String> = emptyList(),
    val userId: String,
    val isPublic: Boolean = false,
    val coverImageUrl: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "name" to name,
            "description" to description,
            "recipeIds" to recipeIds,
            "userId" to userId,
            "isPublic" to isPublic,
            "coverImageUrl" to coverImageUrl,
            "createdAt" to createdAt,
            "updatedAt" to updatedAt
        )
    }
    
    companion object {
        fun fromMap(map: Map<String, Any?>, id: String): Cookbook {
            return Cookbook(
                id = id,
                name = map["name"] as? String ?: "",
                description = map["description"] as? String,
                recipeIds = (map["recipeIds"] as? List<*>)?.mapNotNull { it as? String } ?: emptyList(),
                userId = map["userId"] as? String ?: "",
                isPublic = map["isPublic"] as? Boolean ?: false,
                coverImageUrl = map["coverImageUrl"] as? String,
                createdAt = map["createdAt"] as? Long ?: System.currentTimeMillis(),
                updatedAt = map["updatedAt"] as? Long ?: System.currentTimeMillis()
            )
        }
    }
}

File: ios/Zest/Models/Cookbook.swift
import Foundation

struct Cookbook: Codable, Identifiable {
    var id: String?
    let name: String
    let description: String?
    let recipeIds: [String]
    let userId: String
    let isPublic: Bool
    let coverImageUrl: String?
    let createdAt: Date
    let updatedAt: Date
    
    init(id: String? = nil, name: String, description: String? = nil, recipeIds: [String] = [], userId: String, isPublic: Bool = false, coverImageUrl: String? = nil) {
        self.id = id
        self.name = name
        self.description = description
        self.recipeIds = recipeIds
        self.userId = userId
        self.isPublic = isPublic
        self.coverImageUrl = coverImageUrl
        self.createdAt = Date()
        self.updatedAt = Date()
    }
    
    func toDict() -> [String: Any] {
        return [
            "name": name,
            "description": description as Any,
            "recipeIds": recipeIds,
            "userId": userId,
            "isPublic": isPublic,
            "coverImageUrl": coverImageUrl as Any,
            "createdAt": Timestamp(date: createdAt),
            "updatedAt": Timestamp(date: updatedAt)
        ]
    }
    
    static func fromDict(_ dict: [String: Any], id: String) -> Cookbook? {
        guard let name = dict["name"] as? String,
              let userId = dict["userId"] as? String else {
            return nil
        }
        
        return Cookbook(
            id: id,
            name: name,
            description: dict["description"] as? String,
            recipeIds: dict["recipeIds"] as? [String] ?? [],
            userId: userId,
            isPublic: dict["isPublic"] as? Bool ?? false,
            coverImageUrl: dict["coverImageUrl"] as? String
        )
    }
}
